### PR TITLE
Some CI improvements + Ghidra 11 compilation

### PR DESCRIPTION
**Pull request by antoniovazquezblanco**
Simplify the pipelines a bit.
Fix Ghidra 11 compilation.

Beware I have not tested the release part although it is very similar to others I am currently using in my projects...

### CODE REVIEW

File: src/main/java/cartographer/CartographerPlugin.java [hunk 1]
```diff
@@ -67,7 +67,7 @@
 @PluginInfo(
     status = PluginStatus.RELEASED,
     packageName = MiscellaneousPluginPackage.NAME,
-    category = PluginCategoryNames.DECOMPILER,
+    category = PluginCategoryNames.COMMON,
```
Comment (datalocaltmp) [inline src/main/java/cartographer/CartographerPlugin.java:70]: You may want to set this to PluginCategoryNames.DIAGNOSTIC which would align with the other two pull requests which address this and other issues

Comment (antoniovazquezblanco) [reply src/main/java/cartographer/CartographerPlugin.java]: Thanks for the feedback. Easy fix :)

**Comment by antoniovazquezblanco** [comment]
Beware that the DIAGNOSTIC category may not be available for mersions prior to 11.1 and that may break compatibility with those. I will keep the DIAGNOSTIC catergory and previous versions in CI but you may want to take a decision on not supporting older versions or to fix the code in other way.

I will await instructions :)

**Comment by antoniovazquezblanco** [comment]
Thanks!

**Pull request closed without merging**